src/mbgl/actor/scheduler.cpp

```diff
@@ -1,22 +1,36 @@
 #include <mbgl/actor/scheduler.hpp>
-#include <mbgl/util/thread_local.hpp>
 #include <mbgl/util/thread_pool.hpp>
 
-namespace mbgl {
+#include <mbgl/util/thread_local.hpp>
 
-util::ThreadLocal<Scheduler> g_currentScheduler;
+namespace mbgl {
 
+#ifdef THREAD_LOCAL_SUPPORTED
+namespace {
+    thread_local Scheduler* currentScheduler;
+} // namespace
+#else
+util::ThreadLocal<Scheduler> currentScheduler;
```

Comment:
Instead of having these ifdefs, could the `util::ThreadLocal` encapsulate this and just use `thread_local` so that we have the same API?

Reply:
Unfort. it cannot be a class member

Reply:
I think @kkaefer meant that instead of having ifdefs on the client side it would be nicer to keep them inside `util::ThreadLocal` implementation.

Reply:
this is done, I moved it into the ThreadLocal implementation